README.md

```diff
@@ -1,105 +1,179 @@
+
+1. Make sure you already have a cStor Pool Created or you can 
+   create one using the below command. In the below spc.yaml make sure 
+   that maxPools should be greater than or equal to the number of 
+   replicas required for the volume.
+
+   The following command will create the specified number of cStor Pools
+   using the Sparse files. 
+
+   ```
+   kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/spc.yaml
+   ```
+
+2. Create a Storage Class to dynamically provision volumes 
+   using OpenEBS CSI provisioner. A sample storage class looks like:
+   ```
+   kind: StorageClass
+   apiVersion: storage.k8s.io/v1
+   metadata:
+     name: openebs-csi-cstor-sparse
+     namespace: kube-system
+     annotations:
+   provisioner: openebs-csi.openebs.io
+   allowVolumeExpansion: true
+   parameters:
+     storagePoolClaim: cstor-sparse-pool
+     replicaCount: "1"
+   ```
+   You will need to specify the correct cStor SPC from your cluster 
+   and specify the desired `replicaCount` for the volume. The `replicaCount`
+   should be less than or equal to the max pools available.  
+
+   The following file helps you to create a Storage Class
+   using the cStor sparse pool created in the previous step. 
+   ```
+   kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/sc.yaml
+   ```
+
+3. Run your application by specifying the above Storage Class for 
+   the PVCs. 
+
+   The following example launches a busybox pod using a cStor Volume 
+   provisioned via CSI Provisioner. 
+   ```
+   kubectl apply -f https://raw.githubusercontent.com/openebs/csi/master/deploy/busybox-csi-cstor-sparse.yaml
+   ```
+
+   Verify that the pods is running and is able to write the data. 
+   ```
+   $ kubectl get pods
+   NAME      READY   STATUS    RESTARTS   AGE
+   busybox   1/1     Running   0          97s
+   ```
+
+   The busybox is instructed to write the date when it starts into the 
+   mounted path at `/mnt/openebs-csi/date.txt`
+
+   ```
+   $ kubectl exec -it busybox -- cat /mnt/openebs-csi/date.txt
+   Wed Jul 31 04:56:26 UTC 2019
+   ```
+   
+
+### How does it work?
+
+The following steps indicate the PV provisioning workflow as it passes
+through various components. 
+
+1. Create PVC with Storage Class referring to OpenEBS CSI Driver.
+
+2. Kubernetes will pass the PV creation request to the OpenEBS
+   CSI Controller service via `CreateVolume()`, as this controller
+   registered with Kubernetes for receiving any requests related to
+   `openebs-csi.openebs.io`  
+
+3. OpenEBS CSI Controller will create a custom resource called 
+   CStorVolumeClaim(CVC) and returns the details of the newly 
+   created object back to Kubernetes. The cStorVolumeClaim will be
+   monitored by the cstor-operator (embedded in m-apiserver). The
+   cstor-operator will process the request once the Kubernetes 
+   determines the node on which application using the PVC should 
+   be scheduled.
```

Comment:
The cstor-operator will wait for process the request until the application pods using the PVC has been scheduled on a node ?

Reply:
reworded a bit. please check.